.github/workflows/build.yml

```diff
@@ -150,10 +150,15 @@ jobs:
                     echo "Changes detected"
                     git status
                     git add .
-                    git diff > changes.diff
                     exit 1
                 fi
 
+            # If there are changes, capture the changes and upload them as an artifact
+            - name: Capture changes
```

Comment:
Doing this as a separate task helps us identify issues more easily in CI

Reply:
Yeah for sure, the --staged flags ensures we're only diffing the changes detected, meaning those we staged in the previous step. By separating it into its own step instead of including it above, the logs are now easier to read. The more command line steps you run together, the harder it is to distinguish the results in the log output.